src/researchhub_document/views/researchhub_unified_document_views.py

```diff
@@ -475,8 +476,21 @@ def get_filtered_queryset(
             )
         elif filtering == "-created_date":
             qs = qs.order_by(filtering)
-        elif filtering == "-hot_score":
-            qs = qs.order_by("-hot_score_v2")
+        elif filtering == "-hot_score" or not filtering:
+            featured_doc_ids = FeaturedContent.objects.filter(hub_id=hub_id).values(
+                "unified_document"
+            )
+
+            if featured_doc_ids.count() > 0:
+                featured_docs = ResearchhubUnifiedDocument.objects.filter(
+                    id__in=featured_doc_ids
+                )
+                qs = qs.exclude(id__in=featured_docs.values_list("id", flat=True))
+                qs = qs.order_by("-hot_score_v2")
+                qs = list(featured_docs) + list(qs)
```

Comment:
We want to preserve the ordering of the original queryset and append `featured_docs` to the front. After multiple attempts with `union` and `|`, this is the only thing that worked. Confirmed that pagination and result set is proper.

Answer borrowed from https://stackoverflow.com/questions/18235419/how-to-chain-django-querysets-preserving-individual-order

Reply:
Thanks @le-sun, I will benchmark this and see what queries this thing generates.
LMK if you have any other ideas here.